tests/pr_test.go

```diff
@@ -54,6 +57,43 @@ func TestRunBasicExample(t *testing.T) {
 	assert.NotNil(t, output, "Expected some output")
 }
 
+func TestRunAdvanceExample(t *testing.T) {
+	t.Parallel()
+
+	options := testschematic.TestSchematicOptionsDefault(&testschematic.TestSchematicOptions{
+		Testing: t,
+		Prefix:  "advanced-key-protect",
+		TarIncludePatterns: []string{
+			"*.tf",
+			"scripts/*.sh",
+			"examples/fscloud/*.tf",
+			"modules/*/*.tf",
+			"kubeconfig/README.md",
+		},
+		ResourceGroup:          resourceGroup,
+		TemplateFolder:         advancedExampleTerraformDir,
+		Tags:                   []string{"test-schematic"},
+		DeleteWorkspaceOnFail:  false,
+		WaitJobCompleteMinutes: 60,
+	})
+
+	// If "jp-osa" was the best region selected, default to us-south instead.
+	// "jp-osa" is currently not allowing hs-crypto be used for encrypting in that region.
+	if options.Region == "jp-osa" {
+		options.Region = "us-south"
+	}
+
```

Comment:
There is currently a provider bug which this test will hit when using private only instance. To workaround it, please add this:

```
// Workaround for https://github.com/IBM-Cloud/terraform-provider-ibm/issues/5154
options.AddWorkspaceEnvVar("IBMCLOUD_KP_API_ENDPOINT", "https://private."+region+".kms.cloud.ibm.com", false, false)
```